napari/view_layers.py

```diff
@@ -112,20 +114,24 @@ def _merge_layer_viewer_sigs_docs(func):
 _dims_params = Dims.__fields__
 
 
-def _make_viewer_then(add_method: str, args, kwargs) -> Viewer:
-    """Utility function that creates a viewer, adds a layer, returns viewer."""
+def _make_viewer_then(
+    add_method: str, args, kwargs, viewer=None
+) -> Tuple[Viewer, Any]:
+    """Utility function that creates a viewer, adds a layer, returns viewer
+    and layer."""
```

Comment:
no harm in adding this here, but it could also contain a note that this is the function we might be killing as we get rid of `view_*`...   And when the last `view_*` function is removed, then the body of this function would just move below to `imshow` (which would be the only remaining function to use this logic)